types/express-session/index.d.ts

```diff
@@ -31,6 +31,13 @@ declare global {
              * Even though this property isn't marked as optional, it won't exist until you use the `express-session` middleware
              */
             sessionID: string;
+
+            /**
+             * The Store in use.
+             * Even though this property isn't marked as optional, it won't exist until you use the `express-session` middleware
+             * The function `generate` is added by express-session
+             */
+            sessionStore: session.Store & { generate: (req: express.Request) => void };
```

Comment:
Maybe I'm just tired and confused, but perhaps it would be good to extract this to a separate type alias?

Reply:
Sorry I don't quite understand, do you mean like 
`
type generateFunc = (req: express.Request) => void
`
and 
`
sessionStore: session.Store & { generate: generateFunc };
`
or another way?

Reply:
Honestly I'm not entirely sure what I mean, [this](https://github.com/expressjs/session/blob/1010fadc2f071ddf2add94235d72224cf65159c6/index.js#L158-L166) is one of the weirder pieces of JS I've encountered. I don't know what the idiomatic way would be to model a method that rudely modifies an input object, implicitly coupling it to its new "container" (the session), like some sort of reverse composition.

Regardless of questionable code, this "augmented" `Store`-type might trickle down to other places, so I think it'd be good to have a neat little alias for it. This also reduces the implictness of the weird coupling.

I was thinking something along the lines of:
```ts
type CoupledSessionStore = session.Store & { generate: (req: express.Request) => void };
```
```ts
interface Request {
    sessionStore: CoupledSessionStore;
}
```
...but possibly with a better name 😅 

@peterblazejewicz Any thoughts on this?

Reply:
Yes, that would help. also just 'SessionStore' to be self-exlanatory, maybe?